base/math.jl

```diff
@@ -1202,9 +1202,13 @@ end
 rem2pi(x::Float32, r::RoundingMode) = Float32(rem2pi(Float64(x), r))
 rem2pi(x::Float16, r::RoundingMode) = Float16(rem2pi(Float64(x), r))
 rem2pi(x::Int32, r::RoundingMode) = rem2pi(Float64(x), r)
-function rem2pi(x::Int64, r::RoundingMode)
+function rem2pi(x::Integer, r::RoundingMode)
+    fx = Float64(x)
+    fx == x || throw(ArgumentError("Integer argument to rem2pi is too large: $x"))
+    rem2pi(fx, r)
+end
+function rem2pi(x::Real, r::RoundingMode)
     fx = Float64(x)
-    fx == x || throw(ArgumentError("Int64 argument to rem2pi is too large: $x"))
```

Comment:
I feel like we need a similar check --- if the conversion is inaccurate the answer might be completely wrong.

Reply:
Thanks.  Based on your comment and some additional thinking on my part, I have added an additional commit to this PR: ee83f2ff9b4d39af1dd9721d3857f9ee9a4a2745.  I'm not sure that it actually addresses the fundamental issue you raise, but let me explain my reasoning.

When I first opened this PR, I thought it made sense to convert to a `Float64`, but I have since realized that calling `float` will work better more generally.  This was motivated by thinking about how `BigFloat` should behave, even though it turns out that `rem2pi(::BigFloat, r)` [is already defined in mpfr.jl](https://github.com/JuliaLang/julia/blob/23b474301b15fba3948419b2b11ed3c89eb90726/base/mpfr.jl#L729), so this is not strictly needed for `BigFloat`.  (Note that `rem2pi` [is already defined](https://github.com/JuliaLang/julia/blob/23b474301b15fba3948419b2b11ed3c89eb90726/base/math.jl#L1202-L1203) for `Float32` and `Float16` just above the current block of code.)

Although this change does not result in a similar check to the integer case, it now puts `rem2pi` on the same footing as the `::Real` implementation of [many other functions](https://github.com/JuliaLang/julia/blob/23b474301b15fba3948419b2b11ed3c89eb90726/base/math.jl#L1310-L1318), including trig functions which, like this one, accept radians as argument.  I don't see any reason why the bar for accuracy should be higher for `rem2pi` than for those functions (think of `tan`, for instance, which has the same period but is more poorly conditioned because of the asymptotes).